src/test/java/pro/trevor/tankgame/Rule/BuyActionWithGoldTest.java

```diff
@@ -0,0 +1,137 @@
+package pro.trevor.tankgame.Rule;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import pro.trevor.tankgame.rule.definition.player.PlayerActionRule;
+import pro.trevor.tankgame.rule.impl.shared.PlayerRules;
+import pro.trevor.tankgame.state.State;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.unit.Tank;
+
+public class BuyActionWithGoldTest 
```

Comment:
None of these tests have tanks with existing actions.

Reply:
I agree, I will add a couple of cases.

Reply:
I was also thinking it would be good to have a test case where it doesn't go over the Max AP (like start with 4, try to buy 3), but that would require adding the other rule and I'm not sure how I feel about testing the interactions here.

Reply:
Yeah, I considered asking for a test that goes over max actions but that's more of a test of generic test and the invariant.  It might make sense as a rule set level integration test though.